Backend/Backend/Infrastructure/FileStorageHelper.cs

```diff
@@ -0,0 +1,296 @@
+using static System.IO.Directory;
+using static System.IO.Path;
+using static System.IO.File;
+
+namespace Backend.Infrastructure;
+
+/// <summary>
+/// Handles File interactions to/from the server.
+/// </summary>
+public static class FileStorageHelper
+{
+
+    /// <summary>
+    /// Seperator key for the file name and file type from the filename stored in the db.
+    /// </summary>
+    public const string FILE_SPLIT_KEY = "|/|\\|";
+
+    /// <summary>
+    /// Base dir where all server files are stored.
+    /// </summary>
+    public const string BASE_DIR = "ServerFiles";
+
+    /// <summary>
+    /// Dir where all user avatars are stored.
+    /// </summary>
+    public const string AVATARS_DIR_NAME = "Avatars";
+
+    /// <summary>
+    /// Dir where all podcast covers and episodes are stored. 
+    /// </summary>
+    public const string PODCASTS_DIR_NAME = "Podcasts";
+
+    #region User Profile
+
+    /// <summary>
+    /// Saves a user avatar and returns the filename stored in the database.
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <param name="avatarFile"></param>
+    /// <returns></returns>
+    public static string SaveUserAvatar(Guid userId, IFormFile avatarFile)
+    {
+        return SaveUserAvatar(userId.ToString(), avatarFile);
+    }
+
+    /// <summary>
+    /// Saves a user avatar and returns the filename stored in the database.
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <param name="avatarFile"></param>
+    /// <returns></returns>
+    public static string SaveUserAvatar(string userId, IFormFile avatarFile)
+    {
+        // Filename stored on the server filesystem
+        string avatarFileName = string.Format("{0}.{1}", userId, avatarFile.ContentType.Split('/')[1]);
+        
+        // Filename stored in the database
+        string userAvatarName = string.Format("{0}{1}{2}", avatarFileName, FILE_SPLIT_KEY, avatarFile.ContentType);
+
+        // Get the dir path
+        string dirPath = Combine(GetCurrentDirectory(), BASE_DIR, AVATARS_DIR_NAME);
+
+        // Make sure that the dir exists, otherwise create it
+        if(!Directory.Exists(dirPath))
+            CreateDirectory(dirPath);
+
+        // Get the file path
+        string filePath = Combine(dirPath, avatarFileName);
+
+        // Save the file
+        using FileStream fileStream = Create(filePath);
+        avatarFile.CopyTo(fileStream);
+
+        // Return the filename stored in the database
+        return userAvatarName;
+    }
+
+    /// <summary>
+    /// Removes a user avatar.
+    /// </summary>
+    /// <param name="userAvatarName"></param>
+    public static void RemoveUserAvatar(string userAvatarName)
```

Comment:
Could this potentially delete a placeholder one? like the default placeholder one

Reply:
No, the placeholder would not be saved on the server.